var/spack/repos/builtin/packages/arrayfire/package.py

```diff
@@ -17,27 +17,27 @@ class Arrayfire(CMakePackage, CudaPackage):
 
     license("FreeImage")
 
-    version("master")
-    version("3.8.1", commit="823e8e399fe8c120c6ec7ec75f09e6106b3074ca", tag="v3.8.1")
-    version(
-        "3.7.3", commit="59ac7b980d1ae124aae914fb29cbf086c948954d", submodules=True, tag="v3.7.3"
-    )
-    version(
-        "3.7.2", commit="218dd2c99300e77496239ade76e94b0def65d032", submodules=True, tag="v3.7.2"
-    )
-    version(
-        "3.7.0", commit="fbea2aeb6f7f2d277dcb0ab425a77bb18ed22291", submodules=True, tag="v3.7.0"
-    )
+    with default_args(submodules=True, no_cache=True):
+        version("master")
+        version("3.9.0", commit="b59a1ae535da369db86451e5b28a7bc0eaf3e84a", tag="v3.9.0")
+        version("3.8.1", commit="823e8e399fe8c120c6ec7ec75f09e6106b3074ca", tag="v3.8.1")
+        version("3.7.3", commit="59ac7b980d1ae124aae914fb29cbf086c948954d", tag="v3.7.3")
+        version("3.7.2", commit="218dd2c99300e77496239ade76e94b0def65d032", tag="v3.7.2")
+        version("3.7.0", commit="fbea2aeb6f7f2d277dcb0ab425a77bb18ed22291", tag="v3.7.0")
 
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
 
     variant("forge", default=False, description="Enable graphics library")
     variant("opencl", default=False, description="Enable OpenCL backend")
+    variant("cuda", default=True, description="Enable CUDA backend")
 
     depends_on("boost@1.70:")
     depends_on("fftw-api@3:")
     depends_on("blas")
+    depends_on("fmt@8.1.1:")
+    depends_on("spdlog@1.9.2:")
```

Comment:
Using:
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/fmt-11.0.2-7mdjkkqwuba65yegusdoperbdeettrfo
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/spdlog-1.14.1-d4biprq36z5iuavhdjthbh7rrf3tx

Building arrayfire-3.9.0 using cmake-3.16.9 and gcc-8.4.0
```py
  >> 116    CMake Error at CMakeLists.txt:262 (add_library):
     117      add_library cannot create ALIAS target "af_spdlog" because target
     118      "spdlog:spdlog_header_only" is imported but not globally visible.
```

arrayfire-3.8.1 with cmake-3.16.9 and spdlog-1.14.1 failed with:

```py
  >> 315    /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/spdlog-1.14.1-d4biprq36z5iuavhdjthbh7rrf3txecb/include/spdlog/fmt/fmt.h:24:14: fatal error: spdlog/fmt/bundled/core.h: No such file or directory
     316       24 |     #include <spdlog/fmt/bundled/core.h>
     317          |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
     318    compilation terminated.
```

Those may be triggered by the pkg-config tool possibly missing.
If that helps, it should be added using `depends_on()`.

arrayfire-3.7.3 does not support gcc-13 and likely same for clang-14:
```py
     1295    /.stage/spack-stage-arrayfire-3.7.3-dxdl4bqtndhktjvwztdjazes2scvbooh/spack-src/src/backend/common/half.hpp:1184:26: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
  >> 1296    /.stage/spack-stage-arrayfire-3.7.3-dxdl4bqtndhktjvwztdjazes2scvbooh/spack-src/src/backend/common/half.hpp:1184:34: error: template argument 1 is invalid
     1297     1184 |         return std:hash<uint16_t>()(
```